src/google-maps/google-map/google-map.ts

```diff
@@ -246,10 +244,30 @@ export class GoogleMap implements OnChanges, OnInit, OnDestroy {
     }
   }
 
-  ngOnChanges() {
-    this._setSize();
-    if (this.googleMap && this.mapTypeId) {
-      this.googleMap.setMapTypeId(this.mapTypeId);
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['height'] || changes['width']) {
+      this._setSize();
+    }
+
+    const googleMap = this.googleMap;
```

Comment:
All things being equal, I think we may as well just continue to reference this.googleMap instead of reassigning it to a local variable.

Reply:
It's a bit of a micro-optimization, but I wrote it because this way it'll minify better with most common minifiers that don't do property renaming.

Reply:
Can you add a comment saying that was the reason for doing this?